### PR TITLE
Ticket34/change password - with some backend code remaining

### PR DESCRIPTION
**Pull request by VijayIyer**
### Description
_Please search Ticket 34 in the following document to get details of this ticket - [link to all tickets]_(https://docs.google.com/document/d/1cGjJw3XpMK6vGA0GNbpmBgRsRoXWFVJKLnGnpC_DgWc/edit?pli=1)

This pull request adds changes for **Ticket 34 - Change Password (full stack)**. It adds all the relevant frontend and backend code for the **Change Password** component in the **Settings** Page of the application. From a user's perspective, it gives them a chance to change the saved password for their profile. 

In this pull request, the following new files are included - 
* `pages/settings` -> Settings page inside which Change Password and other settings components will be rendered
* `components/ChangePasswordForm` -> Directory with code, styles and unit tests for components inside ChangePasswordForm.  
   * `ChangePasswordButton`
   * `PasswordTextField` - TextField MUI component with some styles specific to the application
   * `components/common/TimedAlert` - component which is rendered in the **Settings** page but controlled by successful submission on Change Password form
* `pages/api/auth/change-password` -> request handler for change password PUT request

Some new methods are included in the **`lib/userLib/userApi.ts`** file - `getUserById` , **`lib/userLib/comparePassword`**, comparePassword might need to be removed after another method was added in a different PR `comparePasswordWithEmail`

### Notes
* Incomplete items for this ticket
    * **Unit testing for backend code is still to be added**
    * In the unit tests for the Change Password form component, there is 1 failing test case
    * The backend code makes the following assumption : When user sends a request to update password, they send a Bearer Token, which is set in the cookies part of the request header. When the request is sent, then the middleware is supposed to use the token to get the userId and set that in request header for the request handler to use
    * Some new changes might happen to **pages/api/change-password** request handler method
        * backend needs some more code in the change-password handler - _invalidating token on password update_, which would require a new field in the User model (tokenIssuedAt)
* Possible refactoring required in all the submitted files
* **Some files included in this unwantedly** : `EntryPageHeader.tsx`, `package-lock.json` : seems to be an issue related to single quotes becoming double quotes on save

Even though some items are incomplete, all submitted can be reviewed and commented upon for following standard coding guidelines, conventions and principles.

### CODE REVIEW

**Comment by VijayIyer** [comment]
@PHANTOMS0308 @zhiyuan-Z though I have noted the above Pull request needs some modifications or new Pull request with modifications will be added, the code itself should be possible to review and comment on.

**Comment by VijayIyer** [comment]
I have added some more backend code in the last few commits

File: lib/jwtLib/generateToken.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import jwt from "jsonwebtoken";
+type UserAuthDetails = {
+  email: string;
+  id: string;
+};
+export default function generateJwtToken({ id, email }: UserAuthDetails) {
+  const JWT_SECRET = process.env.JWT_SECRET;
+  if (!JWT_SECRET) return null;
```
Comment (zhiyuan-Z) [inline lib/jwtLib/generateToken.ts]: Maybe you could throw an error here

File: lib/userLib/comparePassword.ts
Comment (zhiyuan-Z) [inline lib/userLib/comparePassword.ts:1]: I think we actually already have a comparePasswords utility function in userLib/hashPassword.ts if you pull the latest dev branch.

**Review by zhiyuan-Z — APPROVED** [review]
Great job!

File: lib/jwtLib/generateToken.ts (hunk 1 above)
Comment (VijayIyer) [reply lib/jwtLib/generateToken.ts]: Made the change here. throwing an error, which will get caught by the try catch around the calling method

File: lib/userLib/comparePassword.ts
Comment (VijayIyer) [reply lib/userLib/comparePassword.ts:1]: Removed redundant method - comparePasswords

**Review by PHANTOMS0308 — APPROVED** [review]

**Pull request merged**